vpn/tunnel.go

```diff
@@ -0,0 +1,229 @@
+package vpn
+
+import (
+	"context"
+	"database/sql/driver"
+	"encoding/json"
+	"fmt"
+	"io"
+	"reflect"
+	"strconv"
+	"sync"
+	"unicode"
+
+	"golang.org/x/xerrors"
+
+	"cdr.dev/slog"
+)
+
+type Tunnel struct {
```

Comment:
Note to reviewers:

This component is not finished and not unit tested. I built the basic skeleton of it to inform how the API for the `speaker` should look, and as an illustrative example for reviewers of how the `speaker` will be used in practice.

It will be completed and tested after we have the other pieces in place to actually start the CoderVPN tunnel using the `tailnet` package.